documentation/SolutionsRegistryFormat.md

```diff
@@ -0,0 +1,196 @@
+
+**Example blocks**:
+```
+"configure": [
+    "settings.myconf"
+],
+"restore": [
+    "settings.myconf"
+],
+"start": [
+    {
+        "type": "gpii.launch.exec",
+        "command": "\"${{registry}.HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\nvda.exe\\}\""
+    }
+],
+"stop": [
+    {
+        "type": "gpii.windows.killProcessByName",
+        "filename": "nvda.exe"
+    }
+]
+```
+
+### update
+
+The `update` block works very similarly to the configure, restore, start and stop blocks. It describes what should happen when the configuration needs to be updated (eg. due to context changes, PCP adjustments, etc).
+
+The format of the `update` block allows for the same entries as the configure, restore, start and stop blocks - that is: arbitrary lifecycle action blocks and `settings.<blockname>`. Unlike for the other lifecycle blocks, the `update` block furthermore allows references to the `start`, `stop` and `configure` blocks. This is one by putting a string with the name of that block. When the system encounters one of these references, the entries of that block will be run.
+
+**Example block**:
+```
+"configure": [
+    "settings.myconf"
+],
+"update": [
+    "configure",
+    {
+        "type": "gpii.launch.exec",
+        "command": "my_application --refresh"
+    }
+]
+```
+
+In the above example, the process of updating the application settings would consists of running the contents of the `configure` block (that is `"settings.myconf"`), followed by a custom lifecycle actions.
+
+
+### isInstalled:
+
+This directive is used to detect whether a solution is installed. If any of these blocks evaluate to `true` (implicit **OR**), the application is considered to be installed.
+
+**Example Entry**:
+```
+"isInstalled": [
+    {
+        "type": "gpii.reporter.fileExists",
+        "fileName": "${{registry}.HKEY_CURRENT_USER\\Software\\Texthelp\\Read&Write10\\InstallPath}\\ReadAndWrite.exe"
+    }, // SEEMS LIKE IMPLICIT OR BETWEEN THESE BLOCKS
+    {
+        "type": "gpii.packageKit.find",
+        "name": "orca"
+    }
+]
+```
+
+*****
+
+### UNIMPLEMENTED BLOCKS
+There are several advanced options that we're not planning to implement in the short term, but which will make the implementation of things like the ORCA settings handler much less horrible.
+
+#### isRunning:
+To detect whether a solution is running - this is planned to be integrated in the relatively short run. If any of these blocks evaluate to `true` (implicit OR), the application is considered to be running.
+
+**Example Entry**:
+```
+"isRunning": [
+    {
+        "type": "gpii.processReporter.byName",
+        "fileName": "firefox.exe"
+    },
+    {
+       "type": "gpii.processReporter.find",
+       "command": "ps aux | grep 'myprocess | wc -l'",
```

Comment:
@kaspermarkus,
A number of comments:
First, I'm not clear the intent of the "command" property here.  Is it temporary and just for testing?  Or is that how you envision the actual gpii.processReporter working?

Compare with what I'm doing in my patch with a "type"/"name" pair (I'm fine with using "command" instead of "name").  The "type" specifies the processReporter function, and the "name" is the name of the process to find.  The stuff at the following link means, "look for a process named 'orca'":
https://github.com/klown/universal/blob/GPII-442/testData/solutions/linux.json#L438

Secondly, I think the single quote characters should be around "myprocess" only, that is:
"ps aux | grep 'myprocess' | wc -l"

Thirdly, that will return 1 even if myprocess is not running since it will match the grep itself.  Here is what I get substituting "orca" for "myprocess", i.e. "ps aux | grep 'orca' | wc -l":
* With orca running, outputs 2. 
* With orca off, outputs 1.

Note that running the command without piping to "wc -l", when orca is not running gives a result of:
"clown     4570  0.0  0.2 113000  2376 pts/0    S+   12:41   0:00 grep --color=auto orca"